### DISCUSSION TITLE
About auto sync

### DISCUSSION THREAD

**Discussion by LazyCountry** [remotely-save/remotely-save#32] [Ideas]
- Because there is a technically impossible, can't auto sync while Obsidian is in background
- Obsidian can know which file opened,  which files relate opened file, which file edit

We can provide the full sync ability, then user open a file, we just check this file from the cloud service,  if has newer, pull this quickly. then check relately files, if need pull them.
If user edit the file, push auto.

If support above features, can handle most situations.

**Comment by exoticknight** [comment]
what if user begins to edit when pulling is still in progress?

**Reply by LazyCountry** [reply]
Yes, it is an issue, if the file is big. Maybe need show syncing progress for this file.
I think this plugin is for personal, at most time, I knew I changed the file at other side, And I want pull this file quickly, then edit this.

**Comment by fyears** [comment]
Answered in https://github.com/remotely-save/remotely-save/issues/39
